src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1176,15 +1188,29 @@ def join(
             eq = Column(left_col).equal_null(Column(right_col))
             on = eq if on is None else on & eq
 
-        # If we are doing a cross join, `on` cannot be specified.
-        if how != "cross":
+        if how == "asof":
+            left_match_condition = Column(left_match_condition)
+            # Get the new mapped right match condition identifier
+            right_match_condition = Column(
+                right_identifiers_rename_map[right_match_condition]  # type: ignore
```

Comment:
how come we need a type ignore here?

Reply:
I added an `assert right_match_col` earlier in the function, but mypy still complains ` Invalid index type "Optional[str]", expected type "str"`

Reply:
you will need to add it right before this line